sql/sql_parse.cc

```diff
@@ -9080,8 +9079,25 @@ kill_one_thread(THD *thd, longlong id, killed_state kill_signal, killed_type typ
         thd->security_ctx->user_matches(tmp->security_ctx))
 #endif /* WITH_WSREP */
     {
+#ifdef WITH_WSREP
+      DEBUG_SYNC(thd, "before_awake_no_mutex");
+      if (tmp->wsrep_killed)
+      {
+        /* victim is in hit list already, bail out */
+	WSREP_DEBUG("victim has wsrep_killed set, skipping awake()");
+	error = 0;
+      }
+      else
+      {
+#endif /* WITH_WSREP */
+      WSREP_DEBUG("kill_one_thread %llu, victim: %llu wsrep_killed %d by signal %d",
+                  thd->thread_id, id, tmp->wsrep_killed, kill_signal);
       tmp->awake_no_mutex(kill_signal);
+      WSREP_DEBUG("victim: %llu taken care of", id);
       error=0;
+#ifdef WITH_WSREP
+      }
+#endif /* WITH_WSREP */
```

Comment:
Indent with TABs, not spaces. What is protecting `tmp->wsrep_killed` here?

Please avoid excessive `#ifdef`:
```c++
#ifdef WITH_WSREP
if (…)
{
…
}
else
#endif
{
…
}
```

Reply:
find_thread_by_id() acquires LOCK_thd_data before this